Dev docs: Improve formatting of linux debug tips (#56181)
JuliaLang/julia: doc/src/devdocs/build/linux.md

## Linux Build Troubleshooting

> For OpenBLAS build failures these are some possible solutions.

Reviewer: ```suggestion
If you are building OpenBLAS from source (`USE_BINARYBUILDER_OPENBLAS=0`) and it is failing to compile, these are some possible solutions.
```

In your `Make.user` file set one of the following and build again:

- Set `OPENBLAS_TARGET_ARCH` to `BARCELONA` or `NEHALEM`, respectively for AMD and Intel CPUs.
- Set `OPENBLAS_DYNAMIC_ARCH` to `0` to disable support for multiple targets in a single library.
- Set `OPENBLAS_NO_AVX2` to `1` to allow for disabling AVX for older binutils.
- Set `USE_SYSTEM_BLAS` to `1` to use a system provided `libblas`.
  - Use `LIBBLAS=-l<library>` and `LIBBLASNAME=<name>` if multiple BLAS versions are installed.

If `OPENBLAS_TARGET_ARCH` is set, we default to disabling `OPENBLAS_DYNAMIC_ARCH`.

If you get an error that looks like ```../kernel/x86_64/dgemm_kernel_4x4_haswell.S:1709: Error: no such instruction: `vpermpd $ 0xb1,%ymm0,%ymm0'```, then you need to set `OPENBLAS_DYNAMIC_ARCH = 0` or `OPENBLAS_NO_AVX2 = 1`, or you need a newer version of `binutils` (2.18 or newer). ([Issue #7653](https://github.com/JuliaLang/julia/issues/7653))

> If the linker cannot find `gfortran` and you get an error like `julia /usr/bin/x86_64-linux-gnu-ld: cannot find -lgfortran`, check the path with `gfortran -print-file-name=libgfortran.so` and use the output to export something similar to this: `export LDFLAGS=-L/usr/lib/gcc/x86_64-linux-gnu/8/`. See [Issue #6150](https://github.com/JuliaLang/julia/issues/6150#issuecomment-37546803).

Reviewer: ```suggestion
If the linker cannot find `libgfortran` and you get an error like `julia /usr/bin/x86_64-linux-gnu-ld: cannot find -lgfortran`, check the path with `gfortran -print-file-name=libgfortran.so` and use the output to export something similar to this: `export LDFLAGS=-L/usr/lib/gcc/x86_64-linux-gnu/8/`. See [Issue #6150](https://github.com/JuliaLang/julia/issues/6150#issuecomment-37546803).
```

> For an illegal instruction error check if your CPU supports AVX while your OS does not (e.g. through virtualization, as described in [this issue](https://github.com/JuliaLang/julia/issues/3263)).

Reviewer: ```suggestion
For an illegal instruction error check if your CPU supports AVX while your OS does not (e.g. through virtualization, as described in [issue #3263](https://github.com/JuliaLang/julia/issues/3263)).
```
